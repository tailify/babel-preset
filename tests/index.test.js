'use strict';

const babel = require('@babel/core');

it('should not contain invalid rules', () => {
  const result = babel.transform('let b = 1; a ** b;', {
    babelrc: false,
    presets: ['./lib/index.js'],
    sourceType: 'script',
  });

  expect(result).toBeTruthy();
  expect(result.code).toBe('var b = 1;\nMath.pow(a, b);');
});
