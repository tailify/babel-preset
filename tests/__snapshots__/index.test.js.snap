// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compile targeting browsers 1`] = `
"var b = 1;
Math.pow(a, b);"
`;

exports[`should compile targeting the current node version by default 1`] = `
"let b = 1;
a ** b;"
`;
